rewriter/Visibility.cc

```diff
@@ -0,0 +1,46 @@
+#include "rewriter/Visibility.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Context.h"
+#include "core/Names.h"
+#include "core/core.h"
+#include "core/errors/rewriter.h"
+#include "rewriter/rewriter.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+void Visibility::run(core::MutableContext ctx, ast::ClassDef *classDef) {
+    core::NameRef currentVisibility = core::Names::public_();
+
+    for (auto &stat : classDef->rhs) {
+        typecase(
+            stat.get(), [&](ast::Assign *assign) {},
+            [&](ast::Send *send) {
+                if (send->args.size() != 0 && send->recv != nullptr) {
+                    return;
+                }
+                // Technically this node can now be removed, but it doesn't really need to be removed now
+                if (send->fun == core::Names::private_()) {
+                    currentVisibility = core::Names::private_();
+                } else if (send->fun == core::Names::protected_()) {
+                    currentVisibility = core::Names::protected_();
+                } else if (send->fun == core::Names::public_()) {
+                    currentVisibility = core::Names::public_();
+                }
+            },
+            [&](ast::MethodDef *mdef) {
+                if (currentVisibility == core::Names::private_()) {
+                    ast::MethodDef::ARGS_store newArgs;
+                    newArgs.reserve(0);
+                    auto privateCall = ast::MK::Send1(mdef->loc, ast::MK::Self(mdef->loc), core::Names::private_(),
+                                                      ast::MK::Symbol(mdef->loc, mdef->name));
```

Comment:
```suggestion
                    auto privateCall = ast::MK::Send1(mdef->declLoc, ast::MK::Self(mdef->declLoc), core::Names::private_(),
                                                      ast::MK::Symbol(mdef->declLoc, mdef->name));
```

`MethodDef::loc` is the whole method def, from the `def` keyword all the way to the `end` keyword at the end of the body. It usually isn't great to show in error messages. We probably want `declLoc` here, which is just from the `def` to the end of the line